milabench/pack.py

```diff
@@ -306,6 +325,56 @@ def install(self):
             if reqs.exists():
                 self.pip_install("-r", reqs)
 
+    def pin(self, *pip_compile_args, requirements_file=None,
+            input_files:list=tuple(), constraints:list=tuple(),
+            with_mb:bool=True, cwd=None):
+        """Pin versions to requirements file.
+
+        Arguments:
+            *pip_compile_args: `python3 -m piptools compile` extra arguments
+            requirements_file: The output requirements file
+            input_files: A list of inputs to piptools compile
+            constraint: The constraint file
+            with_mb: Compute the requirements with milabench as a dependency
+            cwd: The working directory
+        """
+        if requirements_file is None:
+            requirements_file = self.requirements_file
+        if requirements_file is None:
+            return
+
+        requirements_file = XPath(requirements_file)
+        if cwd is None:
+            cwd = self.pack_path
+        constraints = tuple(XPath(c).absolute() for c in constraints)
+
+        if cwd != self.pack_path:
+            for fn in (*input_files, requirements_file,
+                       requirements_file.with_suffix(".in")):
+                if fn and (self.pack_path / fn).exists():
+                    (self.pack_path / fn).copy(cwd / fn)
+
+        if (cwd / requirements_file).with_suffix(".in").exists():
+            input_files = (*input_files, requirements_file.with_suffix(".in"))
+
+        import milabench
+        milabench_pyproject = XPath(milabench.__file__).parent.parent / "pyproject.toml"
+        if with_mb and milabench_pyproject.exists():
+            input_files = (*input_files, milabench_pyproject)
```

Comment:
I'm not entirely sure we want to do this. It's useful to avoid requirements.txt files which make milabench crash but at the same time these files will need to be updated more often to following the dependencies update of the project